packages/zowe-explorer/src/utils/ProfilesUtils.ts

```diff
@@ -316,43 +314,44 @@ export class ProfilesUtils {
         ZoweLogger.trace("ProfilesUtils.writeOverridesFile called.");
-        let fd: number;
-        let fileContent: string;
+        const defaultImperativeJson = { overrides: { CredentialManager: globals.ZOWE_CLI_SCM } };
         const settingsFile = path.join(getZoweDir(), "settings", "imperative.json");
+        let fileContent: string;
         try {
-            fd = fs.openSync(settingsFile, "r+");
-            fileContent = fs.readFileSync(fd, "utf-8");
-        } catch {
-            // If reading the file failed because it does not exist, then create it
-            // This should never fail, unless file system is read-only or the file
-            // was created by another process after first openSync call
-            fd = fs.openSync(settingsFile, "wx");
+            fileContent = fs.readFileSync(settingsFile, { encoding: "utf-8" });
+        } catch (error) {
+            ZoweLogger.debug(localize("writeOverridesFile.readFile.error", "Reading imperative.json failed. Will try to create file."));
         }
-        try {
-            let settings: any;
-            if (fileContent) {
-                try {
-                    settings = JSON.parse(fileContent);
-                } catch (err) {
-                    if (err instanceof Error) {
-                        throw new Error(
-                            localize("writeOverridesFile.jsonParseError", "Failed to parse JSON file {0}:", settingsFile) + " " + err.message
-                        );
-                    }
-                }
-                if (settings && settings?.overrides && settings?.overrides?.CredentialManager !== globals.PROFILE_SECURITY) {
-                    settings.overrides.CredentialManager = globals.PROFILE_SECURITY;
-                } else {
-                    return;
+        let settings: any;
+        if (fileContent) {
+            try {
+                settings = JSON.parse(fileContent);
+                ZoweLogger.debug(localize("writeOverridesFile.readFile", "Reading imperative.json Credential Manager.\n {0}", fileContent));
+            } catch (err) {
+                if (err instanceof Error) {
+                    const errorMsg = localize("writeOverridesFile.jsonParse.error", "Failed to parse JSON file {0}: {1}", settingsFile, err.message);
+                    ZoweLogger.error(errorMsg);
+                    throw new Error(errorMsg);
                 }
-            } else {
-                settings = { overrides: { CredentialManager: globals.PROFILE_SECURITY } };
             }
-            fileContent = JSON.stringify(settings, null, 2);
-            fs.writeFileSync(fd, fileContent, "utf-8");
-        } finally {
-            fs.closeSync(fd);
+            if (settings?.overrides?.CredentialManager === globals.PROFILE_SECURITY) {
+                return;
+            }
+            if (!settings?.overrides?.CredentialManager) {
+                settings = { ...defaultImperativeJson, ...settings };
```

Comment:
I believe we want to replace the settings instead of spreading them over the `defaultImperativeJson`.
Otherwise, we are never changing the setting back once it's `false`
```typescript
            if (!settings?.overrides?.CredentialManager) {
                settings = { ...settings, ...defaultImperativeJson };
            }
```